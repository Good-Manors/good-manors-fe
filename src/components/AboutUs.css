@import "compass/css3";

.AboutUs {
  //colors
  $white: #ffffff;
  $de_york_approx: rgba(124,191,144,1);
  $black_10: rgba(0,0,0,0.1);
  $oxley_approx: #72a882;
  $mine_shaft_approx: #333;
  $mine_shaft_70_approx: rgba(51,51,51,0.7);
  $storm_dust_80_approx: rgba(102,102,102,0.8);
  $white_90: rgba(255,255,255,0.8);

  //fonts
  $font_0: Palatino Linotype;
  $font_1: Book Antiqua;
  $font_2: Palatino;
  $font_3: serif;
  $font_4: Arial;
  $font_5: Helvetica;
  $font_6: sans-serif;

  @mixin imgAttach {
    background-repeat: no-repeat; background-position: center;
  }

  #wrap {
    width: 960px;
  }

  .chOpacity {
    opacity: 1 !important;
  }

  .chWhiteText {
    h3 {
      color: #FFFFFF !important;
      a {
        color: #FFFFFF !important;
        text-decoration: none !important;
      }
    }
    
    p {
      color: #FFFFFF !important;
      padding: 1px 0 !important;
      margin: 0 28px !important;
      a {
        color: #FFFFFF !important;
        text-decoration: none !important;
        text-transform: none !important;
        font-weight: 400 !important;
        font-size: 16px !important;
      }
    }
  }
  .chGrid {
    margin: 20px 0 0;
    padding: 0;
    list-style: none;
    display: block;
    text-align: center;
    width: 100%;
    &:after {
      content: '';
      display: table;
      clear: both;
    }
    li {
      width: 200px;
      height: 200px;
      display: inline-block;
      margin: 10px;
	}
}
  .chItem {
    width: 100%;
    height: 100%;
    border-radius: 0%;
    position: relative;
    cursor: default;
    transition: all 0.4s ease-in-out;
    &:before {
      content: '';
      display: table;
    }
    &:hover {
      box-shadow: inset 0 0 0 110px $white_90, 0 1px 2px $black_10;
      .ch-info {
        opacity: 1;

      }
    }
  }

  .chInfo {
    position: absolute;
    width: 100%;
    height: 100%;
    border-radius: 50%;
    overflow: hidden;
    opacity: 0;
    transition: opacity 0.4s ease-in-out;
    backface-visibility: hidden;

    h3 {
      color: $mine_shaft_approx;
      position: relative;
      font-size: 20px;
      margin: 0 4px;
      padding: 70px 0 0;
      height: 36px;
      font-family: $font_0, $font_1, $font_2, $font_3;
    }

    p {
      color: $mine_shaft_approx;
      padding: 10px 4px;
      margin: 0 30px;
      font-size: 14px;
      border-top: 1px solid $mine_shaft_approx;
      font-family: $font_4, $font_5, $font_6;

      a {
        display: block;
        color: $mine_shaft_70_approx;
        font-style: normal;
        font-weight: 700;
        text-transform: uppercase;
        font-size: 9px;
        letter-spacing: 1px;
        padding-top: 4px;
        font-family: $font_4, $font_5, $font_6;
        &:hover {
          color: $storm_dust_80_approx;
        }
      }
    }
  }

  .chOpacityBg:hover {
    box-shadow: inset 0 0 0 110px $de_york_approx, 0 1px 2px $black_10 !important;
  }

  .chInfoLinewhite p {
    border-top-color: #FFFFFF !important;
  }

  .chImg-1 {
    background-image: url(../assets/about/Angela.png);
    @include img_attach{};
  }

  .chImg-2 {
    background-image: url(../assets/about/Allison.jpg);
    @include img_attach{};
  }

  .chImg-3 {
    background-image: url(../assets/about/Luke.jpg);
    @include img_attach{};
  }

  .chImg-4 {
    background-image: url(../assets/about/John.jpg);
    @include img_attach{};
  }
}
