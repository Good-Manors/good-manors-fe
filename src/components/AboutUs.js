import React from 'react';
import styles from './AboutUs.css';

export default function AboutUs() {
  return (
    <section className={styles.AboutUs}>
      <h1>Meet our team</h1>

      <ul className={styles.chGrid}>
        <li>
          <div className={styles.chItem}>
            <div className={styles.chInfo, styles.chImg1}>
              <h3>
                Angela Vawser</h3>
              <p>description</p>
              <h5>hello-jelly.com</h5>
            </div>
          </div>
        </li>

        <li>
          <div className={styles.chItem}>
            <div className={styles.chInfo, styles.chImg2}>
              <h3>
                Allison Busse</h3>
              <p>description</p>
              <h5>allisonbusse.com</h5>
            </div>
          </div>
        </li>

        <li>
          <div className={styles.chItem}>
            <div className={styles.chInfo, styles.chImg3}>
              <h3>
                Luke Donahue</h3>
              <p>description</p>
              <h5>github.com/Luke9389</h5>
            </div>
          </div>
        </li>

        <li>
          <div className={styles.chItem}>
            <div className={styles.chInfo, styles.chImg4}>
              <h3>
                John Nelson</h3>
              <p>description</p>
              <h5>github.com/John4850</h5>
            </div>
          </div>
        </li>
      </ul>
    </section>

  );
}
