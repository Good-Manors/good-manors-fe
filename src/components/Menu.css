.Menu {
  grid-area: menu;
  display: flex;
  flex-direction: row;
  align-content: center;
  justify-content: flex-end;
  margin: 0 35px 0 0;
  width: 100%;

  h5 {
    font-family: 'Sulphur', sans-serif;
    font-weight: 700;
    font-size: 1rem;
    color: black;
    letter-spacing: .08em;
    margin: 38px 10px;
  }

  a {
    font-family: 'Sulphur', sans-serif;
    font-weight: 700;
    font-size: .8rem;
    color: black;
    letter-spacing: .08em;
    padding: 8px 0;
    margin: 0 7px;
    text-decoration: none;
  }

  a:focus {
    outline: none;
  }

}

:global(.bm-burger-button) {
  position: relative;
  width: 26px;
  height: 20px;
  margin: 35px 30px 0 0;
}

:global(.bm-burger-bars) {
  background: black;
}

:global(.bm-burger-bars-hover) {
  background: #8ab4b1;
}

:global(.bm-cross-button) {
  height: 24px;
  width: 24px;
}

:global(.bm-cross) {
  background: black;
}

:global(.bm-cross-hover) {
  background-color: #8ab4b1;
}

:global(.bm-menu-wrap) {
  position: fixed;
  height: 100%;
}

:global(.bm-menu) {
  padding: 2.5em 1.5em 0;
  font-size: 1.15em;
}

:global(.bm-item-list) {
  color: #b8b7ad;
  padding: 0.8em;
}

:global(.bm-item) {
  display: inline-block;
}

:global(.bm-overlay) {
  background: rgba(0, 0, 0, 0);
  display: none;
}
