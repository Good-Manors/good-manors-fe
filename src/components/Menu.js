import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { slide as HamburgerMenu } from 'react-burger-menu';
import styles from './Menu.css';

const mapStateToProps = (state) => ({
  username: state.username
});

class hamburgerMenu extends React.Component {
  showSettings(event) {
    event.preventDefault();
  }

  constructor(props) {
    super(props);
    this.state = {
      menuOpen: false
    };
  }

  handleStateChange(state) {
    this.setState({ menuOpen: state.isOpen });
  }

  closeMenu() {
    this.setState({ menuOpen: false });
  }

  render() {
    return (
      <>
        <div className={styles.Menu}>
          <h5>hi, {this.state.username}</h5>
          <HamburgerMenu right width={'200px'} isOpen={this.state.menuOpen} onStateChange={(state) => this.handleStateChange(state)}>
            <Link to="/" onClick={() => this.closeMenu()} className={'menu-item'}>Dashboard</Link>
            <Link to="/about-us" onClick={() => this.closeMenu()} className={'menu-item'}>About Us</Link>
          </HamburgerMenu>
        </div>
      </>
    );
  }
}

hamburgerMenu.propTypes = {
  handleSubmit: PropTypes.func.isRequired
};

export default connect(mapStateToProps)(hamburgerMenu);
