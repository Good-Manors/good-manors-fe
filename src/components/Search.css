.Search {
  grid-area: search;
  margin: 0 2em;
  
  div {
    display: flex;
    border-radius: 50%;
    height: 44px;
    justify-content: center;
    margin-top: 20px;
  }

  h2 {
    text-align: center;
    color: black;
    font-size: 2.4em;
  }

  img {
    width: 28px;
    height: 28px;
    background-color: #8ab4b1;
    padding: 6px;
    border-top-left-radius: 25px;
    border-bottom-left-radius: 25px;

  }

  input {
    border-top-right-radius: 25px;
    border-bottom-right-radius: 25px;
    height: 29px;
    width: 80%;
    margin: 0;
  }
}
