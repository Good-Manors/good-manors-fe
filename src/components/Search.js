import React from 'react';
import PropTypes from 'prop-types';
import styles from './Search.css';
import searchIcon from '../assets/icons/search-icon.png';

export default function Search({ home, setDrawersOpen, setSearchTerm }) {
  let title;
  if(home) title = home.title;

  return (
    <section className={styles.Search}>
      <h2>{title || ''}</h2>
      <div>
        <img src={searchIcon} />
        <input type='text' onChange={({ target }) => {
          setSearchTerm(target.value);
          setDrawersOpen(true);
        }} />
      </div>
    </section>
  );
}

Search.propTypes = {
  home: PropTypes.object
};

