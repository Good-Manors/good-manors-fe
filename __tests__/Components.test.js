import React from 'react';
import { mount, shallow } from 'enzyme';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import Search from '../src/components/Search';
import Menu from '../src/components/Menu';
import AboutUs from '../src/components/AboutPage';
import UserForm from '../src/components/user/UserForm';
import Dashboard from '../src/components/user/Dashboard';

describe('Tests small components', () => {
  
  it('Search.js', () => {
    const wrapper = shallow(<Search />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Menu.js', () => {
    jest.mock('react-redux', () => ({
      useDispatch: ()=> { },
      useSelector: () => ({})
    }));
    const getWrapper = (mockStore = createStore()) => mount(
      <Provider store={mockStore}>
        <Menu />
      </Provider>
    );
    expect(getWrapper).toMatchSnapshot();
  });

  it('AboutUs.js', () => {
    const wrapper = shallow(<AboutUs />);
    expect(wrapper).toMatchSnapshot();
  });

});

describe('User folder', () => {
  it('UserForm.js', () => {
    jest.mock('react-redux', () => ({
      useDispatch: ()=> { },
      useSelector: () => ({})
    }));

    const getWrapper = (mockStore = createStore()) => mount(
      <Provider store={mockStore}>
        <UserForm buttonText={'test'} handleSubmit={()=>{}} />
      </Provider>
    );
    expect(getWrapper).toMatchSnapshot();
  });
 
  it('Dashboard.js', () => {
    const wrapper = shallow(<Dashboard />);
    expect(wrapper).toMatchSnapshot();
  });
  
});
