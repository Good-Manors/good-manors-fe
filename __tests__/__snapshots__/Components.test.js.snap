// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests small components AboutUs.js 1`] = `
<section
  className="AboutUs"
>
  <h1>
    Meet our team
  </h1>
  <ul
    className="chGrid"
  >
    <li>
      <div
        className="chItem"
      >
        <div
          className="chImg1"
        >
          <h3>
            Angela Vawser
          </h3>
          <p>
            description
          </p>
          <h5>
            hello-jelly.com
          </h5>
        </div>
      </div>
    </li>
    <li>
      <div
        className="chItem"
      >
        <div
          className="chImg2"
        >
          <h3>
            Allison Busse
          </h3>
          <p>
            description
          </p>
          <h5>
            allisonbusse.com
          </h5>
        </div>
      </div>
    </li>
    <li>
      <div
        className="chItem"
      >
        <div
          className="chImg3"
        >
          <h3>
            Luke Donahue
          </h3>
          <p>
            description
          </p>
          <h5>
            github.com/Luke9389
          </h5>
        </div>
      </div>
    </li>
    <li>
      <div
        className="chItem"
      >
        <div
          className="chImg4"
        >
          <h3>
            John Nelson
          </h3>
          <p>
            description
          </p>
          <h5>
            github.com/John4850
          </h5>
        </div>
      </div>
    </li>
  </ul>
</section>
`;

exports[`Tests small components Menu.js 1`] = `[Function]`;

exports[`Tests small components Search.js 1`] = `
<section
  className="Search"
>
  <h2 />
  <div>
    <img
      src="/path/image"
    />
    <input
      onChange={[Function]}
      type="text"
    />
  </div>
</section>
`;

exports[`User folder Dashboard.js 1`] = `
<h1>
  Dashboard
</h1>
`;

exports[`User folder UserForm.js 1`] = `[Function]`;
